### ISSUE TITLE
Feature request - register custom events with event manager

### ISSUE THREAD

**Issue by PeterGribbin** [RoboMagus/OctoPrint-Custom_gcode_events#1]
I would like to run a script when gcode 'M141' is sent. If this addon registered the custom events with the event register, and passed the line of gcode to it as an argument, this would be possible.

The event manager offers this hook to do it: https://docs.octoprint.org/en/master/plugins/hooks.html#sec-plugins-hook-events-register-custom-events

**Comment by RoboMagus** [comment]
Hey @PeterGribbin,

Back when I first started work on this plugin I had looked into it, but back then there was a bug in Octoprint that caused inconsistent event id naming (https://github.com/OctoPrint/OctoPrint/issues/4417).
Looking at that issue now it seems that It's been fixed but not yet released. I was planning to hold out till the 1.8.0 release before I implement the event registration to avoid issues down the line.

In the meantime, you can actually subscribe to events that are not registered in that way too.
I've tried the following as a quick demo and the logging shows that it works:

### Config.yaml snippet:
```yaml
events:
  subscriptions:
  - command: logger '====  LCD TEST  ===='
    debug: true
    event: gcode_event_lcd_test
    type: system
```

### Custom GCode events setting:
![image](https://user-images.githubusercontent.com/68224306/166259001-91277bc7-5ffc-4778-8b4b-bedc1f8972a5.png)

### Log output:
```
2022-05-02 17:13:30,879 - octoprint.plugins.custom_gcode_events - INFO - Sent match for 'M117'. Firing event 'gcode_event_lcd_test'
2022-05-02 17:13:30,894 - octoprint.events - INFO - Executing system command: logger '====  LCD TEST  ===='
```

I hope that helps you out.

Lets keep the issue open though because I do want to have the event registration implemented at some point ;)

*EDIT:*
To verify the working of this configuration, I just sent an M117 command to the printer to set the LCD display text using the Octoprint terminal.

**Comment by PeterGribbin** [comment]
Thanks, this worked! I was able to access the full gcode line that triggered the event too.
`- command: print_script.py {__data}`
returns:
`{'gcode': 'M141', 'cmd': 'M141 S20', 'event': 'temp'}`

**Comment by HTF81** [comment]
Hi RoboMagus, I'm wondering if this will be added any time soon? It's exactly the functionality I need to get my project working as intended. I am pulling out what's left of the little hair I have to get this working so I'm hoping you can help.

**Comment by RoboMagus** [comment]
Apologies for the delay @HTF81, as I'm barely interacting with my 3D printing setup lately I kind of forgot about this...

I've got some spare time in the near future so I should be able to work on this then.

**Comment by RoboMagus** [comment]
Out of curiosity though, what is your usecase for using these events that they can only be used when properly registered?

**Comment by RoboMagus** [comment]
@HTF81, I've drafted a pre-release of the implementation for this FR. 
Could you update to this version to see if it suits your needs?

**Comment by HTF81** [comment]
Thank you so much @RoboMagus, this appears to be working perfectly.

**Comment by RoboMagus** [comment]
Nice 👍 

FYI, I'll keep in in pre-release for a bit longer to see if any bugs roll out.

**Comment by Gshaw12** [comment]
> Moved irrelevant question to separate thread.

**Comment by RoboMagus** [comment]
Closing this issue, as the feature is now released in v[0.3.2](https://github.com/RoboMagus/OctoPrint-Custom_gcode_events/releases/tag/0.3.2).
